### PR TITLE
Add notebook with constrained Bayesian optimization example

### PR DESCRIPTION
**Pull request by marpulli**
*Issue #, if available:* #174

*Description of changes:* Notebook showing how the acquisition algebra can be used to enforce non-linear constraints.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: notebooks/Emukit-constrained-optimization.ipynb [hunk 1]
```diff
@@ -0,0 +1,358 @@
+{
```
Comment (apaleyes) [inline notebooks/Emukit-constrained-optimization.ipynb]: <p>Would be nice to break this code down into few cells, with a short text description for each</p>

 _Reply via <a href='https://app.reviewnb.com/amzn/emukit/pull/196/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (apaleyes) [inline notebooks/Emukit-constrained-optimization.ipynb]: <p>A short comment like `expit is scipy's implementation of sigmoid` would be handy</p>

 _Reply via <a href='https://app.reviewnb.com/amzn/emukit/pull/196/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (apaleyes) [inline notebooks/Emukit-constrained-optimization.ipynb]: <p>caled -&gt; called</p>

 _Reply via <a href='https://app.reviewnb.com/amzn/emukit/pull/196/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (apaleyes) [inline notebooks/Emukit-constrained-optimization.ipynb]: <p>This this way of applying constraints allows violations, would be great to have at least a note of what people should do if that happens.</p>

 _Reply via <a href='https://app.reviewnb.com/amzn/emukit/pull/196/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (marpulli) [reply notebooks/Emukit-constrained-optimization.ipynb]: <p>Added in the intro and pointed out this only works with positive acquisition functions</p>

Comment (marpulli) [reply notebooks/Emukit-constrained-optimization.ipynb]: <p>Done</p>

Comment (marpulli) [reply notebooks/Emukit-constrained-optimization.ipynb]: <p>Done</p>

Comment (marpulli) [reply notebooks/Emukit-constrained-optimization.ipynb]: <p>Done</p>

**Pull request merged**